plugins/source/test/client/spec.go

```diff
@@ -11,6 +15,11 @@ type Spec struct {
 
 	// Number of rows to generate (per row of parent) in test_sub_table.
 	NumSubRows *int `json:"num_sub_rows" jsonschema:"minimum=0,default=10"`
+
+	// Required environment variables. The plugin will fail if these are not set
+	// to the correct values. Specified in `key=value` format. Use `key=` to specify
+	// that the environment variable should be not set or empty.
+	RequiredEnv []string `json:"required_env" jsonschema:""`
```

Comment:
`pattern` would work here for json schema, but I'm not sure about the correct escaping.

```suggestion
	RequiredEnv []string `json:"required_env" jsonschema:"pattern=^[^\=]+\="`
```

Maybe. Could also just do `minLength=2` (would still allow empty slice)